tools/src/main/scala/scala/scalanative/optimizer/pass/CfChainsSimplification.scala

```diff
@@ -14,20 +14,22 @@ class CfChainsSimplification(implicit top: Top) extends Pass {
   import CfChainsSimplification._
 
   override def onInsts(insts: Seq[Inst]): Seq[Inst] = {
-    val cfg = ControlFlow.Graph(insts)
+    val cfg    = ControlFlow.Graph(insts)
+    val usedef = UseDef(cfg)
+    val method = MethodInfo(cfg, usedef)
```

Comment:
The code would be a bit more clear if you make `MethodInfo` an implicit argument. This way you don't have to manually pass it around in recursive calls down the chain.

Reply:
I changed it, is that what you had in mind ?

Reply:
Yep, it's much better now! 👍